src/hooks/useStoredState.ts

```diff
@@ -0,0 +1,21 @@
+import { useState, useCallback } from 'react';
+
+// eslint-disable-next-line import/prefer-default-export
+export function useStoredState<T extends string>(key: string, defaultValue: T): [
+    T,
+    (v: T) => void,
+] {
+    const previousValue = localStorage.getItem(key) as T;
+
+    const [value, setValue] = useState<T>(previousValue || defaultValue);
```

Comment:
To set default value on useState, the argument should be a function which will call an expensive computation only once.
getting data from localStorage in every render is not good